modules/auxiliary/gather/bing_subdomains_collector.rb

```diff
@@ -0,0 +1,145 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit4 < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Bing Search Engine Subdomains Collector',
+      'Description' => %q(
+        This module can be used to gather subdomains about a domain
+        from Bing Search Results.
+      ),
+      'Author' => [ 'Nixawk' ],
+      'License' => MSF_LICENSE))
+
+    register_options(
+      [
+        OptString.new('TARGET', [ true, "The target to locate subdomains for, ex: rapid7.com, 8.8.8.8"]),
```

Comment:
`TARGET` has specific a meaning in other parts of the framework. Maybe `DOMAIN_OR_IP`? Not sure what the best thing would be here.